src/services/impl/LeaderboardManager.cpp

```diff
@@ -22,9 +37,9 @@ RA_Leaderboard* RA_LeaderboardManager::FindLB(LeaderboardID nID)
     return nullptr;
 }
 
-void RA_LeaderboardManager::ActivateLeaderboard(const RA_Leaderboard& lb) const
+void LeaderboardManager::ActivateLeaderboard(const RA_Leaderboard& lb) const
 {
-    if (g_bLBDisplayNotification)
+    if (m_pConfiguration.IsFeatureEnabled(ra::services::Feature::LeaderboardNotifications))
```

Comment:
Standard dependency injection - service was stored as a member field.

The service locator pattern can coexist with dependency injection. The two primary benefits of the service locator pattern are that you don't have to pass a service through a hierarchy of classes to get it to the bottom-most usage, and no storage is required in the class instances for the service. The trade-off is that you won't get compile-time errors if dependencies aren't provided.